src/main/java/org/dependencytrack/tasks/BomUploadProcessingTask.java

```diff
@@ -174,13 +177,15 @@ public void inform(final Event e) {
         try (var ignoredMdcProjectUuid = MDC.putCloseable(MDC_PROJECT_UUID, ctx.project.getUuid().toString());
              var ignoredMdcProjectName = MDC.putCloseable(MDC_PROJECT_NAME, ctx.project.getName());
              var ignoredMdcProjectVersion = MDC.putCloseable(MDC_PROJECT_VERSION, ctx.project.getVersion());
-             var ignoredMdcBomUploadToken = MDC.putCloseable(MDC_BOM_UPLOAD_TOKEN, ctx.token.toString())) {
-            processEvent(ctx, event);
+             var ignoredMdcBomUploadToken = MDC.putCloseable(MDC_BOM_UPLOAD_TOKEN, ctx.token.toString());
+             final Handle jdbiHandle = openJdbiHandle()) {
```

Comment:
Unfortunately this has the potential to cause some issues, because we already keep a database connection open for the entire duration of `processBom` (since it's running in a single transaction). By additionally [opening a JDBI handle](https://jdbi.org/#_managing_the_handle_lifecycle_manually), we'll be keeping an additional connection open.

I'd recommend to use `useJdbiTransaction` or `inJdbiTransaction` ad-hoc, whenever you need to update the workflow state. That ensures that the connection used for those operations is released quickly.